README.md

```diff
@@ -85,6 +88,60 @@ public class Startup
 }
 ```
 
+### Customise MongoDB persistence options and _id generation
+
+In addition to `MongoDbIdentifiable` your resource classes are free to use any of the MongoDB driver persistence options or inherit from their own base class.
+
+For example, you could change the example above so that the `Book` resource has string IDs rather than object ids in the DB, but still have them generated server side:
+
+```cs
+public class Book : IIdentifiable<string>
+{
+    // If Id=null generate a random string ID using the MongoDB driver
+    [BsonId(IdGenerator = typeof(StringObjectIdGenerator))]
+    [Attr]
+    public virtual string Id { get; set; }
+
+    // override the attribute name in the db
+    [BsonElement("bookName")]
+    [Attr]
+    public string Name { get; set; }
+
+    // all json:api resources need this
+    [BsonIgnore]
+    public string StringId { get => Id; set => Id = value; }
+}
+```
+
+Resources just need to inherit from the base `IIdentifiable<string>` interface from JsonApiDotNetCore (or the provided default `MongoDbIdentifiable`) and then just use any of usual [MongoDB Driver mapping code](https://mongodb.github.io/mongo-csharp-driver/2.12/reference/bson/mapping/).
+
+You could also achieve the exact same result using MongoDB `BsonClassMap` [rather than attributes](https://mongodb.github.io/mongo-csharp-driver/2.11/reference/bson/mapping/) so your `Book` does not need any MongoDB specific code like below.
+
+```cs
+    // in startup
+    BsonClassMap.RegisterClassMap<Book>(cm =>
+    {
+        cm.AutoMap();
+        cm.MapIdProperty(x => x.Id).SetIdGenerator(StringObjectIdGenerator.Instance);
+        cm.UnmapMember(x=>x.StringId);
+    });
+```
+
+Using `StringObjectIdGenerator` above could then be combined with `AllowClientGeneratedIds` JsonApi setting in `Startup.ConfigureServices` so that IDs can be generated on the client, and will be auto-assigned server side if not provided providing a flexible string based id for the `Book` resource:
+
+```cs
+    services.AddJsonApi(options => {
+        // Allow us to POST books with already assigned IDs!
+        options.AllowClientGeneratedIds = true;
+    }, resources: builder =>
+    {
+        builder.Add<Book, string>();
+    });
+    services.AddJsonApiMongoDb();
+
+    services.AddResourceRepository<MongoDbRepository<Book, string>>();
+```
+
```

Comment:
I think this is basic JsonApiDotNetCore functionality, which is already documented at https://www.jsonapi.net/usage/options.html#client-generated-ids. I don't see why it needs to be mentioned here, or am I missing something that makes this work _differently_ for MongoDB that requires explanation? Note we have [MongoDB-tests](https://github.com/json-api-dotnet/JsonApiDotNetCore.MongoDb/blob/08b6231649d6d7d5119af6e42f2d4bd3e12e3cf9/test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Creating/CreateResourceWithClientGeneratedIdTests.cs) for this and all seem to work fine without any special handling.

Reply:
The combination of mongoDB features (string ids generated server side) and json api features (optional client side id generation) working in tandem was hard to discover for me and took me a few passes to figure out which was why I was offering some docs up to save others the same hassle but accept what you say.

I think it is MongoDB specific because the library is defaulted to the (performant) binary object id. If you are trying to do client side id generation with AllowClientGeneratedIds then users (me!) may assume they can pass any string id since that is how the API appears. However, in reality the API requires you to pass a 12 byte unique [generated](https://gist.github.com/solenoid/1372386) object id, and throws an non-obvious exception if you just pass something like `"123"`.

Using mongoDB `StringObjectIdGenerator` and json:api `AllowClientGeneratedIds` are a nice fit together in the document db ecosystem because it is common to have client-assigned document random string keys as the default that can be overridden with a unique string if I know the id I want to assign. This differs from relational SQL ecosystem where auto-increment database-assigned ids are the default, and client side key assignment is the exception.

Reply:
Makes sense. But then we should have tests backing that scenario, if we're going to advise on doing that. Can you add some? They would demonstrate using the fluent mapping overrides too, which is nice.